src/compiler/checker.ts

```diff
@@ -30962,7 +30961,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     }
 
     function isHyphenatedJsxName(name: string | __String) {
-        return stringContains(name as string, "-");
+        return (name as string).includes("-");

```

Comment:
```suggestion
        return name.includes("-");
```

The type assertion isn't needed anymore; it was likely only here as the function only accepted `string`, but the method will be present on both `string` and `__String`.

Reply:
I tried it, but it does not compile:

![image](https://github.com/microsoft/TypeScript/assets/2021355/ebcf785f-67b3-44d0-bede-2e7c66d1161c)

Reply:
Agh, yes, I had some editor delay. Shame.